### ISSUE TITLE
Asymmetry with how payin and payout are represented across messages

### ISSUE THREAD

**Issue by mistermoe** [TBD54566975/tbdex#260]
* Offering has: `payinCurrency`, `payoutCurrency`, `payinMethods`, `payoutMethods`
* RFQ has: `payinAmount`, `payinMethod`, `payoutMethod`
* Quote has everything tucked into `payin` and `payout`

Wondering if we should follow what we've done for quote in Offering as well. Concretely,

```
payinCurrency + payinMethods -> payin {
    currencyCode -> currency,
    minAmount -> min,
    maxAmount -> max,
    methods
}

payoutCurrency + payoutMethods -> payout {
    currencyCode -> currency,
    minAmount -> min,
    maxAmount -> max,
    methods
}
```

**Comment by mistermoe** [comment]
decided on:

## Offering
```
payinCurrency + payinMethods -> payin {
    currencyCode -> currency,
    minAmount -> min,
    maxAmount -> max,
    methods
}

payoutCurrency + payoutMethods -> payout {
    currencyCode -> currency,
    minAmount -> min,
    maxAmount -> max,
    methods
}
```

## RFQ
```
type SelectedPayinDetails {
    amount: string
    kind: string
    details: yoloany
}

type SelectedPayoutDetails {
    kind: string
    details: yoloany
}
```

which would end up looking like:

```
{
    payin: {
        amount: "30.00",
        kind: "IDK",
        details: {}
    },
    payout: {
        kind: "IDK",
        details: {}
    }
}
```

Also: change `Quote.currencyCode` -> `Quote.currency`

**Comment by selfissued** [comment]
The improved symmetry looks good to me.

**Comment by angiejones** [comment]
any reason on currencyCode -> currency? The 'code' part of that gave me exact expectations for what to provide in that field without needing to go read the API reference.

**Comment by mistermoe** [comment]
@angiejones fair call out. I may be a bit tunnel visioned here because i couldn't think of what else people would think the value of this field could be aside from a currency code. 

also was swayed by IDEs / code editors rendering the api reference as i type. tried this for kt, js, go, dart. though to be fair, this is just my setup

<img width="1004" alt="image" src="https://github.com/TBD54566975/tbdex/assets/4887440/943ac0a5-d6fd-40fd-a3fd-3c1c30c66948">

**Comment by decentralgabe** [comment]
I agree with your reasoning @mistermoe though I want to point our [currency code](https://www.iban.com/currency-codes) is an industry term that may be worth aligning with.

My only other thought is on the contents of `details` should it be `data`?

**Comment by mistermoe** [comment]
good call @decentralgabe . let's go with `currencyCode`
